### PR TITLE
[SMALLFIX] Minor changes in several files

### PR DESCRIPTION
**Pull request by HorizonNet**
- Used automatic resource management in RemoteBlockInStream
- Fixed link in Javadoc in UnderStoreBlockInStreamTest
- Used <> generics in IndexedSet
- Removed standalone Javadoc snippet in AlluxioException
- Simplified check in DataNettyBuffer

### CODE REVIEW

File: core/client/src/main/java/alluxio/client/block/RemoteBlockInStream.java [hunk 1]
```diff
@@ -139,15 +139,12 @@ private int readFromRemote(byte[] b, int off, int len) throws IOException {
     int bytesLeft = toRead;
     while (bytesLeft > 0) {
       // TODO(calvin): Fix needing to recreate reader each time.
-      RemoteBlockReader reader = RemoteBlockReader.Factory.create();
-      try {
+      try (RemoteBlockReader reader = RemoteBlockReader.Factory.create()) {
         ByteBuffer data = reader.readRemoteBlock(mWorkerInetSocketAddress, mBlockId, getPosition(),
-            bytesLeft, mLockId, mBlockWorkerClient.getSessionId());
+                bytesLeft, mLockId, mBlockWorkerClient.getSessionId());
```
Comment (jsimsa) [inline core/client/src/main/java/alluxio/client/block/RemoteBlockInStream.java]: undo this formatting change

**Comment by jsimsa** [comment]
@HorizonNet thanks for the PR, left one minor comment, otherwise LGTM

**Comment by HorizonNet** [comment]
@jsimsa Thanks for the comment. Another project messed with my settings. I updated the PR accordingly.

**Comment by jsimsa** [comment]
LGTM

**Pull request merged**